### PR TITLE
Add get_local_mem_per_thread method to Dispatcher

### PR DESCRIPTION
**Pull request by testhound**
This pull request partially address issue: https://github.com/numba/numba/issues/6448 by adding support for retrieving the local memory size utilized by a kernel from the Dispatcher.

### CODE REVIEW

**Comment by testhound** [comment]
gpuci run tests

**Review by gmarkall — CHANGES_REQUESTED** [review]
Some quick review points:

* Local memory may be allocated by the compiler or the user, so it's fragile to check for exact local memory usage - instead you need to check that memory usage is greater than or equal to the minimum expected.
* Similarly, `test_get_no_local_memory` is not safe - especially for a GPU with limited resources, spilling to local memory might occur for `func_with_lots_of_registers`.
* Is the level of duplication / copy-pasting starting to get out of hand in test_dispatcher.py? Are there enough commonalities in testing these attributes that we should think about reducing the amount of duplication?
* Please check the spelling in your comments - there are several typos.

**Comment by testhound** [comment]
gpuci run tests

**Comment by testhound** [comment]
@gmarkall I addressed three out of your four requests. After reviewing, I don't agree their is sufficient commonality as the kernels and conditions differ sufficiently.

**Comment by testhound** [comment]
gpuci run tests

File: numba/cuda/tests/cudadrv/test_linker.py [hunk 1]
```diff
@@ -269,6 +284,29 @@ def test_get_shared_mem_per_specialized(self):
         shared_mem_size = compiled_specialized.get_shared_mem_per_block()
         self.assertEqual(shared_mem_size, 800)
 
+    def test_get_no_local_memory(self):
```
Comment (gmarkall) [inline numba/cuda/tests/cudadrv/test_linker.py:287]: I still don't think this test is safe, as the compiler isn't guaranteed to allocate zero local memory even for an empty kernel - similar to the analagous test in https://github.com/numba/numba/pull/8295, I think this should just be removed.

**Review by gmarkall — CHANGES_REQUESTED** [review]
Thanks for the updates. I think the `test_get_no_local_memory` test should be removed, otherwise this looks OK.

File: numba/cuda/tests/cudadrv/test_linker.py (hunk 1 above)
Comment (testhound) [reply numba/cuda/tests/cudadrv/test_linker.py]: @gmarkall test removed.

**Comment by testhound** [comment]
gpuci run tests

**Pull request merged**